bin/import_slr88.py

```diff
@@ -0,0 +1,225 @@
+        )
+        if os.path.isfile(target_csv_template):
+            return
+        else:
+            raise FileNotFoundError
+    except FileNotFoundError:
+
+        all_files = glob(f"{extracted_dir}/{ARCHIVE_NAME_txt}/*.txt")
+
+        transcripts = {}
+        for tr in all_files:
+            with open(tr, "r") as tr_source:
+                transcript = tr_source.read()
+                audio = os.path.basename(tr).replace(".txt", ".wav")
+                transcripts[audio] = transcript
+                tr_source.close()
+
+        # Get audiofile path and transcript for each sentence
+        samples = []
+        for record in glob(f"{extracted_dir}/{ARCHIVE_NAME_wav}/*.wav"):
+            record_file = os.path.basename(record)
+            if record_file in transcripts.keys():
+                samples.append((record, transcripts[record_file]))
+
+        # Keep track of how many samples are good vs. problematic
+        counter = get_counter()
+        num_samples = len(samples)
+        rows = []
+
+        print("Importing WAV files...")
+        pool = Pool()
+        bar = progressbar.ProgressBar(max_value=num_samples, widgets=SIMPLE_BAR)
+        for i, processed in enumerate(pool.imap_unordered(one_sample, samples), start=1):
+            counter += processed[0]
+            rows += processed[1]
+            bar.update(i)
+        bar.update(num_samples)
+        pool.close()
+        pool.join()
+
+        target_csv_template = os.path.join(
+            target_dir, "".join([ARCHIVE_DIR_NAME, "_{}.csv"])
+        )
+
+        with open(
+            target_csv_template.format("train"), "w", encoding="utf-8", newline=""
+        ) as train_csv_file:  # 80%
+            with open(
+                target_csv_template.format("dev"), "w", encoding="utf-8", newline=""
+            ) as dev_csv_file:  # 10%
+                with open(
+                    target_csv_template.format("test"), "w", encoding="utf-8", newline=""
+                ) as test_csv_file:  # 10%
+                    train_writer = csv.DictWriter(train_csv_file, fieldnames=FIELDNAMES)
+                    train_writer.writeheader()
+                    dev_writer = csv.DictWriter(dev_csv_file, fieldnames=FIELDNAMES)
+                    dev_writer.writeheader()
+                    test_writer = csv.DictWriter(test_csv_file, fieldnames=FIELDNAMES)
+                    test_writer.writeheader()
+
+                    for i, item in enumerate(rows):
+                        transcript = validate_label(item[2])
+                        if not transcript:
+                            continue
+                        wav_filename = item[0]
+                        i_mod = i % 10
+                        if i_mod == 0:
+                            writer = test_writer
+                        elif i_mod == 1:
+                            writer = dev_writer
+                        else:
+                            writer = train_writer
```

Comment:
80%/10%/10% split usually leads to large validation and test sets. What we prefer to do in our datasets is a sample size power calculation. See for example `get_sample_size` in `bin/import_fisher.py`.

Reply:
[cd41cc9](https://github.com/coqui-ai/STT/pull/2062/commits/cd41cc9e97fe348d36defe2e81883ff9b1cfc693): It's an half assed job but it should give us the distribution we are looking for.